vue/src/components/UploadModelRun.vue

```diff
@@ -0,0 +1,196 @@
+<script setup lang="ts">
+import { reactive, ref } from "vue";
+import S3FileFieldClient, {
+  S3FileFieldResultState,
+} from "django-s3-file-field";
+import { ApiService } from "../client";
+
+const s3ffClient = new S3FileFieldClient({
+  baseUrl: `${ApiService.getApiPrefix()}/s3-upload/`,
+  apiConfig: {
+    // django csrf token handling
+    xsrfCookieName: "csrftoken",
+    xsrfHeaderName: "X-CSRFToken",
+  },
+});
+
+const modelRun = reactive({
+  title: "",
+  region: "",
+  performer: "",
+  private: false,
+});
+
+const successDialog = ref(false);
+const uploadDialog = ref(false);
+const validForm = ref(false);
+const uploadFile = ref<File>();
+const uploadLoading = ref(false);
+const uploadError = ref<unknown>();
+// Infinity means indeterminate
+const uploadProgress = ref<number>(0);
+
+const QUERY_TASK_POLL_DELAY = 1000; // msec
+// from celery.states.READY_STATES
+const CELERY_READY_STATES = ["FAILURE", "SUCCESS", "REVOKED"];
+
+function openUploadDialog() {
+  uploadDialog.value = true;
+  uploadFile.value = undefined;
+  uploadError.value = undefined;
+}
+
+async function untilTaskReady(taskId: string) {
+  return new Promise((resolve) => {
+    async function queryTask() {
+      const result = await ApiService.getModelRunUploadTaskStatus(taskId);
+
+      if (CELERY_READY_STATES.includes(result)) {
+        resolve(result);
+      } else {
+        setTimeout(() => {
+          queryTask();
+        }, QUERY_TASK_POLL_DELAY);
+      }
+    }
+
+    queryTask();
+  });
+}
+
+async function upload() {
+  try {
+    uploadLoading.value = true;
+    uploadError.value = undefined;
+    uploadProgress.value = Infinity;
+
+    const file = Array.isArray(uploadFile.value)
+      ? uploadFile.value[0]
+      : uploadFile.value;
+    if (!file) return;
```

Comment:
Vuetify is supposed to produce a `File` if `multiple` isn't set, and `File[]` if `multiple` is set (or the modelValue is an array) (https://github.com/vuetifyjs/vuetify/issues/17687). However, I'm consistently seeing an array show up for some unknown reason.

I think it's safer to treat the file ref as being either an array or a file object, given that always returning an array may in fact be buggy behavior. I will also push some changes to properly annotate the ref type, since I have it as `ref<File>()` currently.

Reply:
ahh thanks, I didn't search for an existing bug.